Content.Client/Stunnable/StunSystem.cs

```diff
@@ -1,9 +1,243 @@
+﻿using System.Numerics;
+using Content.Shared.Mobs;
 using Content.Shared.Stunnable;
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Animations;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
 
-namespace Content.Client.Stunnable
+namespace Content.Client.Stunnable;
+
+public sealed class StunSystem : SharedStunSystem
 {
-    public sealed class StunSystem : SharedStunSystem
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly SpriteSystem _spriteSystem = default!;
+
+    private readonly int[] _sign = [-1, 1];
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<StunnedComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<StunnedComponent, MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<StunnedComponent, AppearanceChangeEvent>(OnAppearanceChanged);
+    }
+
+    /// <summary>
+    ///     Add stun visual layers
+    /// </summary>
+    private void OnComponentInit(Entity<StunnedComponent> entity, ref ComponentInit args)
+    {
+        if (_timing.IsFirstTimePredicted)
+            return;
+
```

Comment:
```suggestion
    {
        if (!_timing.IsFirstTimePredicted)
            return;

```
I'm not a prediction expert but shouldn't this be negated so it only runs on first prediction, instead of on every but first?